machine_learning_hep/globalfitter.py

```diff
@@ -58,255 +28,280 @@ def fixpar(massmin, massmax, masspeak, range_signal):
-                tot_fit.GetParameter(4), tot_fit.GetParError(4)))
-        if bkgfunc == "Pol2":
-            pinfom.AddText("%s = %.3f #pm %.3f" % (tot_fit.GetParName(4),\
-                tot_fit.GetParameter(4), tot_fit.GetParError(4)))
-            pinfom.AddText("%s = %.3f #pm %.3f" % (tot_fit.GetParName(5),\
-                tot_fit.GetParameter(5), tot_fit.GetParError(5)))
-        pinfom.Draw()
-        pinfos.AddText("S = %.0f #pm %.0f " % (rawYield, rawYieldErr))
-        pinfos.AddText("B (%.0f#sigma) = %.0f #pm %.0f" % \
-            (nsigma, background, errbackground))
-        pinfos.AddText("S/B (%.0f#sigma) = %.4f " % (nsigma, sigOverback))
-        pinfos.AddText("Signif (%.0f#sigma) = %.1f #pm %.1f " %\
-            (nsigma, significance, errsignificance))
-        pinfos.Draw()
-        c1.Update()
-        c1.SaveAs("%s/fittedplot%s.eps" % (outputfolder, suffix))
+    # Prepare a function to store the signal parameters which will finally be extracted
+    # from the total fit. So this is just a helper for now
+    minForSig = masspeak - 4. * sigma_sig
+    maxForSig = masspeak + 4. * sigma_sig
+    binForMinSig = histo.FindBin(minForSig)
+    binForMaxSig = histo.FindBin(maxForSig)
+    sum_tot = 0.
+    sumback = 0.
+    for ibin in range(binForMinSig, binForMaxSig + 1):
+        sum_tot += histo.GetBinContent(ibin)
+        sumback += back_fit.Eval(histo.GetBinCenter(ibin))
+    integsig = Double((sum_tot - sumback) * (histo.GetBinWidth(1)))
+    sig_fit = TF1("sig_fit", signal_func(sgnfunc), massmin, massmax)
+    sig_fit.SetParameter(0, integsig)
+    sig_fit.SetParameter(1, masspeak)
+    sig_fit.SetParameter(2, sigma_sig)
+    sig_fit.SetParNames(*par_names_sig)
+
+    logger.debug("get and set fit Parameters")
+    par_back1 = back_fit.GetParameters()
+    par_gaus2 = sig_fit.GetParameters()
+
+    # Prepare for total fit by copying parameters from side-band fit and those set
+    # in the signal function
+    pars_tot = array("d", (npar_sig + npar_bkg) * [0.])
+    for ipar in range(0, npar_bkg):
+        pars_tot[ipar] = par_back1[ipar]
+    for ipar in range(npar_bkg, (npar_bkg + npar_sig)):
+        pars_tot[ipar] = par_gaus2[ipar - npar_bkg]
+
+    logger.info("fit all (signal + background)")
+    tot_fit = TF1("tot_fit", tot_func(bkgfunc, massmax, massmin), massmin, massmax)
+    tot_fit.SetLineColor(4)
+    tot_fit.SetParameters(pars_tot)
+    parmin = Double()
+    parmax = Double()
+    for ipar_tot in range(0, npar_sig):
+        sig_fit.GetParLimits(ipar_tot, parmin, parmax)
+        tot_fit.SetParLimits(ipar_tot + npar_bkg, parmin, parmax)
+    if fixedmean:
+        tot_fit.FixParameter(npar_bkg + 1, masspeak)
+    if setfixgaussiansigma is True:
+        tot_fit.FixParameter(npar_bkg + 2, sigma_sig)
+    tot_fit.SetParNames(*par_names_bkg, *par_names_sig)
+    histo.Fit("tot_fit", ("R,%s,+,0" % (fitOption)))
+
+    logger.info("calculate signal, backgroud, S/B, significance")
```

Comment:
After this, `tot_fit` is consistently not used but `sig_fit` and `back_refit` instead where parameters are set to match the ones from `tot_fit`